pdata/pmetric/generated_metrics.go

```diff
@@ -35,19 +35,29 @@ import (
 // Important: zero-initialized instance is not valid for use.
 type ResourceMetricsSlice internal.ResourceMetricsSlice
 
+type MutableResourceMetricsSlice internal.MutableResourceMetricsSlice
```

Comment:
Try this:

```
type ResourceMetricsSlice internal.ResourceMetricsSlice

// This way MutableResourceMetricsSlice inherits all the getters from the non mutable.
type MutableResourceMetricsSlice struct {
    ResourceMetricsSlice
}
```

Reply:
Discussed offline that this has a downside because it allows users to convert between the two types and allowing to convert from immutable to mutable is not intended.